### DISCUSSION TITLE
estimating fee for a Call

### DISCUSSION THREAD

**Discussion by Nick2Nik** [software-mansion/starknet.py#1172] [Ideas]
Maybe I'm wrong - but there is no way to estimate fee for a call now. If I'm wrong - give me an example please. Sorry again, but I lerned documentation and examples.

**Comment by tomek0123456789** [comment]
Hi, in order to estimate `Call` fee, you need to use [`Account.sign_invoke_transaction`](https://starknetpy.readthedocs.io/en/latest/guide/account_and_client.html#creating-transactions-without-executing-them) on the call and the result of that pass into [`Client.estimate_fee`](https://starknetpy.readthedocs.io/en/latest/api/full_node_client.html#starknet_py.net.full_node_client.FullNodeClient.estimate_fee) method, e.g.:
```python
async def estimate_fee():
    client = FullNodeClient(node_url="your_node_url")
    account = Account(
        address="your_account_address",
        client=client,
        key_pair=KeyPair.from_private_key("your_private_key"),
        chain=StarknetChainId.NETWORK_YOU_ARE_USING,
    )
    call = Call(
        to_addr=your_contract_address,
        selector=get_selector_from_name("your_method_name"),
        calldata=[your_calldata],
    )
    invoke = await full_node_account.sign_invoke_transaction(calls=call, max_fee=int(1e16))
    estimated_fee = await full_node_account.client.estimate_fee(tx=invoke)

```
